### PR TITLE
Migration to the Black code style

### LINKED ISSUE #66: Add an automated code style and docstring checker to Travis
Use a lint tool when building the project and also check all docstrings for correct syntax.

### LINKED ISSUE #96: Format code automatically for each pull request
Use a tool like `flake8` or `black` to enforce a consistent code format in the repository.

### PR DESCRIPTION
**Pull request by nathanaelbosch**
Closes #66. The linked issue, and especially [this comment](https://github.com/probabilistic-numerics/probnum/issues/66#issuecomment-659345867), describe the goals and scope of this PR in detail.

### CODE REVIEW

**Comment by nathanaelbosch** [comment]
I would be happy about feedback for the following points:
- Changes to `CONTRIBUTING.md` and `STYLEGUIDE.md`
  I made rather minimal changes here. I did not add any explanation to the "Getting Started" section (I was not sure how/where to do that best), but just to the corresponding "Code Quality" part.
- Do you even feel comfortable with the resulting code?
  Some of it might look unnatural, but I believe that the overall code quality should improve if collaborators configure their IDE or pre-commit correctly, since they would realize how the code looks after formatting and could make adjustments.
- Should I add a code-style check to tox?
  This way, running `tox` locally would check for compliance with *Black*.
  I would however not include that in Travis, I actually find having separate checks visually more appealing.
- Finally: Could we coordinate with all people that actively work on the code to prevent nasty merge conflicts?
  I'm not too sure about the *conflicts that might arise from this merge* or about how to best prevent having any. I believe the worst that could happen would be having to fix a few merge conflicts. I don't have an overview on who is working actively with which parts of the code. In any case, having a second opinion on this topic would be important.

**Comment by nathanaelbosch** [comment]
Comment on the last changes:
In the last commits I added Black to tox.ini, such that GitHub Actions now calles Black via tox.
I also added a `format` option to tox which applies `black`, callable via `tox -e format`. It brings the requirements down to installing (and using) tox and thus makes one-off usage easier. But I still believe the workflow that we want to encourage is to install Black and use it through an IDE and/or through pre-commit.

**Comment by JonathanWenger** [comment]
> I would be happy about feedback for the following points:
> 
>     * Changes to `CONTRIBUTING.md` and `STYLEGUIDE.md`
>       I made rather minimal changes here. I did not add any explanation to the "Getting Started" section (I was not sure how/where to do that best), but just to the corresponding "Code Quality" part.

Since the format of the developer guidelines will change with the new documentation format i think this is perfectly fine for now.

>     * Do you even feel comfortable with the resulting code?
>       Some of it might look unnatural, but I believe that the overall code quality should improve if collaborators configure their IDE or pre-commit correctly, since they would realize how the code looks after formatting and could make adjustments.

For me the code definitely looks better. Even just for identical quotes in the entire repository...
 
>     * Should I add a code-style check to tox?
>       This way, running `tox` locally would check for compliance with _Black_.
>       I would however not include that in Travis, I actually find having separate checks visually more appealing.

Agreed.

>     * Finally: Could we coordinate with all people that actively work on the code to prevent nasty merge conflicts?
>       I'm not too sure about the _conflicts that might arise from this merge_ or about how to best prevent having any. I believe the worst that could happen would be having to fix a few merge conflicts. I don't have an overview on who is working actively with which parts of the code. In any case, having a second opinion on this topic would be important.

Active development is currently going on in `RandomVariable` and `Distribution` by @marvinpfoertner and in `probnum.linalg` by me. I think the earlier we format the repo the better.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -108,3 +109,5 @@ Code coverage of the tests is reported through [codecov](https://codecov.io/gith
 Travis also automatically builds and publishes the [ProbNum documentation](https://probabilistic-numerics.github.io/probnum/modules.html).
 
 Changes to Travis can be made through the `.travis.yml` file, as well as through `tox.ini` since Travis relies on `tox` for both testing and building the documentation.
+
+ProbNum also uses [GitHub Actions](https://docs.github.com/en/actions) to verify that all pushes and pull requests are compliant with the [*Black*](https://github.com/psf/black) code style.
```
Comment (JonathanWenger) [inline CONTRIBUTING.md:113]: I assume in the future we want a reference to pre-commit hooks here?

Comment (nathanaelbosch) [reply CONTRIBUTING.md:113]: I don't have a good idea of how the future docs will look, but I could imagine linking Black's [explanation to set up pre-commit](https://github.com/psf/black#version-control-integration), [pre-commit](https://pre-commit.com/) directly, and/or an explanation on how to set up pre-commit (`pip install pre-commit`, then `pre-commit install`).

In the same context I would also link Black's page on [editor integration](https://github.com/psf/black/blob/master/docs/editor_integration.md).

**Pull request merged**